packages/firestore/src/remote/persistent_stream.ts

```diff
@@ -461,10 +464,10 @@ export abstract class PersistentStream<
     this.state = PersistentStreamState.Backoff;
 
     this.backoff.backoffAndRun(async () => {
-      if (this.state === PersistentStreamState.Stopped) {
-        // Stream can be stopped while waiting for backoff to complete.
-        return;
-      }
+      assert(
```

Comment:
Man am I gun-shy about this assertion. I'm somewhat in favor of just leaving the old behavior in place to keep this consistent with the other callbacks.

Reply:
Yeah...  I am torn between being as strict as possible with our state transitions vs being defensive. I mostly changed it to an assert, since as-is it's basically dead code.

I put it back though and just tweaked the comment.